tests/robot-tests/tests/libs/table_tool.robot

```diff
@@ -22,6 +21,54 @@ user clicks subheaded indicator checkbox
     user clicks element
     ...    xpath://*[@id="filtersForm-indicators"]//legend[text()="${subheading_label}"]/..//label[text()="${indicator_label}"]/../input[@type="checkbox"]
 
+user checks indicator groups match exactly in order
+    [Arguments]    @{items}
+    ${list}=    Get WebElements
+    ...    xpath://*[@id="filtersForm-indicators"]//*[starts-with(@id,"filtersForm-indicators-options-")]//legend
+    ${num_items}=    Get Length    ${items}
+    length should be    ${list}    ${num_items}
+    FOR    ${index}    ${content}    IN ENUMERATE    @{items}
+        user checks element should contain    ${list}[${index}]    ${content}
+    END
+
+user checks indicator group contains exactly items in order
+    [Arguments]    @{items}    ${group_label}
+    ${list}=    Get WebElements    xpath://*[@id="filtersForm-indicators"]//legend[text()="${group_label}"]/..//label
```

Comment:
Feels like it might be handy to hive this off into a keyword that takes an xpath expression to locate the items and the item list, and then reuse it, like:

```
user checks filters match exactly in order
    [Arguments]    @{items}
    user checks items match exactly in order
    ....  xpath://*[@id="filtersForm-filters"]//details
    ....  @{items}

user checks filter contains exactly filter groups in order
    [Arguments]    @{items}    ${filter_label}
    user checks items match exactly in order
    ...  xpath://*[@id="filtersForm-filters"]//details[summary[contains(., "${filter_label}")]]//*[contains(@id,"-options-")]//legend
    ...  @{items}
    

user checks items match exactly in order
    [Arguments]    ${item_selector}  @{expected_items}
    ${list}=    Get WebElements    ${item_selector}
    ${num_items}=    Get Length    ${expected_items}
    length should be    ${list}    ${num_items}
    FOR    ${index}    ${content}    IN ENUMERATE    @{expected_items}
        user checks element should contain    ${list}[${index}]    ${content}
    END
```

Reply:
Oh, or even, you've actually made this already in your keyword `user checks list contains exactly items in order` :)

Can this keyword be used to make these keywords simpler?

Reply:
Thanks, have simplified all these now to use a new `user checks items matching locator contain exact items in order` in `common.robot`. It's similar to the one I'd already written but this one takes a locator such as xpath that's expected to resolve to an array of matching elements, whereas the other one expects the locator to be a list and then gets list items before doing the comparison.